### ISSUE TITLE
[Suggestion] Resume content automatically for a few frames when ejecting disc?

### ISSUE THREAD

**Issue by bslenul** [libretro/beetle-psx-libretro#793]
## Description

I think it would be great if we didn't have to manually resume content after ejecting a disc before inserting a new disc, I feel like it's not super intuitive for most users.

Maybe automatically resume the game (in the background) when ejecting disc? Like just for a few frames until the core "registers" that the disc was ejected? I have no idea how these cores handle that, but resuming is not required with PCSX ReARMed and Duck/SwanStation.

### Expected behavior

No resuming required:

* Go to `Quick Menu > Disc Control` and click "Eject".
* Switch disc index.
* Click "Insert".

### Actual behavior

Resuming is required (maybe not for every game? not sure):

* Go to `Quick Menu > Disc Control` and click "Eject".
* Resume.
* Go back to disc control menu and switch disc index.
* Click "Insert".

### Steps to reproduce the bug

1. Start Metal Gear Solid from disc 2.
2. Select "New game".
3. Try switching disc without resuming, you'll get stuck at the "Insert disc 1" screen.

**Comment by bslenul** [comment]
Hey :)

Nope, I should've said it in the description but I've tested with .m3u, .pbp and just the .cue + "Load New Disc" (AKA append disc), in every case the screen is stuck at "Insert Disc 1", until I eject/resume/insert.

It's not a huge problem when you're aware of it but I've seen a lot of users getting stuck when disc switching is required, thinking they did something wrong, it's really not intuitive, especially when it's not needed with other cores. And I just noticed it but even in the docs the resume part isn't mentioned: https://docs.libretro.com/library/beetle_psx/#swapping-disks

**Comment by ichee** [comment]
Assigning the eject hotkey to a button or key can workaround the issue, instead of opening the menu to eject.

Also, disabling pause while menu is open can also bypass the issue.

**Comment by bslenul** [comment]
Yes there are workarounds, but hotkeys if you're on controller it's a PITA to manage because you may need these buttons for other cores and RetroArch hates hotkeys and overrides (https://github.com/libretro/RetroArch/issues/6641) :p The "Pause When Menu is Active" OFF works fine indeed, I used it for a while (but for some inexplicable reason having the games running in the background bothered me so I turned it back ON... 😅 ).

But yeah the goal would be to make it as user-friendly as possible (especially for new RA users!), so no hotkeys, no extra options, just eject, switch, insert, enjoy :D

**Comment by ichee** [comment]
I don't know much about the 'Disk Control' interface, but maybe it would be possible to toggle 'Pause When Menu is Active' whenever eject is selected.

That should effectively fix the issue.
